nginx_stage/lib/nginx_stage/generators/pun_config_generator.rb

```diff
@@ -95,6 +95,24 @@ class PunConfigGenerator < Generator
       template "pun.conf.erb", config_path
     end
 
+    # Run the pre hook command. This eats the output and doesn't affect
+    # the overall status of the PUN startup
+    add_hook :exec_pre_hook do
+      unless NginxStage.pun_pre_hook.nil?
+        args = ["--user", user]
+        env = {}
+
+        unless STDIN.tty?
+          STDIN.each_line do |line|
+            key_values = line.split('=')
+            env[key_values[0]] = key_values[1] if key_values.size[2]
+          end
+        end
+
+        Open3.capture2e(env, NginxStage.pun_pre_hook, *args)
```

Comment:
We just eat the error. Either we could spit to stderr or tell folks to redirect in the script `2>&1 >/some/prehook/logfile`.